src/Voron/Data/Containers/Container.cs

```diff
@@ -119,11 +120,12 @@ public void PrepareForCommit(Transaction tx)
                     freePages.AddOrSetAfterGetNext(ref key, vals[i]);
                 }
                 
-                index = AddAndSortBuffer(Removals, buffer);
+                index = AddAndSortBuffer(FreeListRemovals, buffer);
```

Comment:
For reference - the fix is here: we passed `Removals` instead of `FreeListRemovals`